main.go

```diff
@@ -0,0 +1,475 @@
+	//  https://developers.google.com/admin-sdk/directory/v1/guides/search-users
+	//	eg. searchFilter := "isAdmin=false"
+
+	searchFilter := ""
+	allUsers, err := findDomainUsers(ctx, *cx, searchFilter, adminService)
+	if err != nil {
+		glog.Errorf("Error finding domain users %v", err)
+		return
+	}
+
+	// Launch a goroutines that will process the projects and organizations
+	// that are visible to users.  The channel is fed by the
+	// getOrganizations(), and getProjects() function calls in the succeeding stanza.
+	ch := make(chan *userAccess)
+	done := make(chan bool)
+	go func() {
+		for {
+			msg, ok := <-ch
+			if ok {
+				if msg.Error != nil {
+					glog.Errorf("Error iterating: %v", msg.Error)
+				} else {
+					for _, o := range msg.Organizations {
+						glog.V(50).Infof("             User %s has Organization visibility to %s", msg.User.PrimaryEmail, o.Name)
+						if _, ok := allOrganizations[o.Name]; !ok {
+							glog.V(2).Infof("             User [%s] has external organization visibility to [%s](%s)", msg.User.PrimaryEmail, o.Name, o.DisplayName)
+						}
+					}
+					for _, p := range msg.Projects {
+						glog.V(50).Infof("             User %s has Project visibility to %s", msg.User.PrimaryEmail, p.ProjectId)
+						if _, ok := allProjects[p.ProjectId]; !ok {
+							glog.V(2).Infof("             User [%s] has external project visibility to [projects/%d](%s)", msg.User.PrimaryEmail, p.ProjectNumber, p.ProjectId)
+						}
+					}
+				}
+			} else {
+				done <- true
+				return
+			}
+		}
+	}()
+
+	// Launch goroutines for each user that feeds a channel received the the go routine defined in the preceding stanza.
+	for _, u := range allUsers {
+
+		// Select a random service account to use per user and disribute quota consumption
+		impersonateAccount, svcKeyCred := getRandomServiceAccount(serviceAccounts, svcAccountKeys)
+
+		// We need to iterate over all organizations visible to the current user
+		// The noFilter is explicitly defined here to allow you to modify which organizations to
+		// limit the query.
+		// see https://pkg.go.dev/google.golang.org/api@v0.58.0/cloudresourcemanager/v1#SearchOrganizationsRequest
+		noFilter := ""
+		wg.Add(1)
+		go getOrganizations(ctx, ch, &wg, limiter, noFilter, impersonateAccount, svcKeyCred, *u)
+
+		//  The Project.List() accepts a Filter parameter which will return a subset
+		//  of projects that match the  specifications.
+		//  By default, if the Filter value is not set, all projects will be returned.
+		//  However, the code below leaves the parameter
+		//  explicitly defined in the event you want to query and evaluate on a
+		//  subset of projects.  See:
+		//  https://pkg.go.dev/google.golang.org/api@v0.58.0/cloudresourcemanager/v1#ProjectsListCall.Filter
+		wg.Add(1)
+		go getProjects(ctx, ch, &wg, limiter, noFilter, impersonateAccount, svcKeyCred, *u)
+		time.Sleep(time.Duration(*delay) * time.Millisecond)
```

Comment:
Could you use the `limiter` here rather than in the `getOrganizations` and `getProjects` functions? Or does it need finer granularity?